libs/sql-parser/src/main/antlr/SqlBase.g4

```diff
@@ -134,8 +134,8 @@ query
 queryNoWith
     : queryTerm
       (ORDER BY sortItem (',' sortItem)*)?
-      (LIMIT limit=parameterOrInteger)?
-      (OFFSET offset=parameterOrInteger)?
+      (LIMIT (limit=parameterOrInteger | nullLiteral | ALL))?
```

Comment:
I think we could extend `parameterOrInteger` with `nullLiteral`. The code should/must already handle null values because a parameter could bind to `null` and it would reduce the amount of special expressions in the grammar

Reply:
It strictly speaking isn't supported from a semantic level, but a user could use a `?` and then pass a `NULL` value, so allowing `NULL` directly in the parser shouldn't break things, as some component after the parser will have to deal with it anyways.

Reply:
ok, let me try with allowing null everywhere.

Reply:
As discussed, I'll merge it with the change to allow `nullLiteral` for `parameterOrInteger` and open another PR to add missing null checks for the `CLUSTERED INTO` and the shard rerouting commands.